pins/drivers.py

```diff
@@ -124,7 +126,15 @@ def load_data(
             try:
                 import rdata
 
-                return rdata.read_rds(f)
+                # Equivalent to `rdata.read_rds(f)` but compatible with Python 3.8.
+                # See https://github.com/rstudio/pins-python/pull/265
+                if isinstance(f, fsspec.implementations.local.LocalFileOpener):
+                    # rdata requires f to be a BinaryIO object.
+                    io_f = f.f
+                else:
+                    io_f = f
+                parsed = rdata.parser.parse_file(io_f)
```

Comment:
```suggestion
                parsed = rdata.parser.parse_file(f.name)
```

Did a bit of digging here-- in the case of rda/rds files, `f` will be a `io.BufferedReader`; the name on this class is the path to the file it is reading. `rdata` is able to handle file paths internally across versions (tried it out locally on 3.8 and seemed to work?), but not the BufferedReader object.

Reply:
Thanks for working that out!